[GTK] Remove support for building against libsoup2

aperezdc opened the pull request:
#### 6153cafaf3bd0db47a50f088e752e73a39d40921
<pre>
[GTK] Remove support for building against libsoup2
<a href="https://bugs.webkit.org/show_bug.cgi?id=300163">https://bugs.webkit.org/show_bug.cgi?id=300163</a>

Reviewed by Michael Catanzaro.

Remove API 4.0 from the GTK port, which is the only remaining one using
libsoup2. This in turn allows to remove all the USE(SOUP2) guards, and
the SOUP_CHECK_VERSION(2, ...) ones. As side effects, the USE_SOUP2
CMake option is gone, and the definition of ENABLE_SERVER_PRECONNECT
is moved into PlatformEnableGlib.h as it is always enabled for both the
WPE and GTK ports.

Canonical link: <a href="https://commits.webkit.org/301131@main">https://commits.webkit.org/301131@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1e8ec62168153a1c0793050bdd80d10fd50d326a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124748 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44420 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35155 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131594 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76667 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126625 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45124 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52990 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94914 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62971 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127702 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35992 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111566 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75483 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34923 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29720 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75070 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/116853 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105749 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29951 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134259 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/123268 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51596 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39405 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103391 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52009 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107786 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103162 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26333 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48530 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26808 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48597 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51469 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57267 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/156321 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50863 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39158 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54218 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52557 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/cmake/OptionsWPE.cmake:
```diff
@@ -165,25 +165,16 @@ if (ENABLE_DOCUMENTATION AND NOT GIDocgen_FOUND)
-if (NOT LibSoup_FOUND)
-    message(FATAL_ERROR "libsoup 3 is required.")
-endif ()
+find_package(LibSoup 3.0.0 REQUIRED)
```
Should move this up to the top of the file.

aperezdc replied:
Done!

mcatanzaro commented on Source/cmake/OptionsGTK.cmake:
```diff
@@ -175,32 +172,12 @@ if (ENABLE_WAYLAND_TARGET AND NOT ${GTK_SUPPORTS_WAYLAND})
-    message(FATAL_ERROR "libsoup 3 is required. Enable USE_SOUP2 to use libsoup 2 (disables HTTP/2)")
-endif ()
-endif ()
+find_package(LibSoup 3.0.0 REQUIRED)
```
Should move this to the top of the file.

aperezdc replied:
Done as well.

mcatanzaro reviewed (approved):
Yay!

aperezdc committed:
[GTK] Remove support for building against libsoup2
https://bugs.webkit.org/show_bug.cgi?id=300163

Reviewed by Michael Catanzaro.

Remove API 4.0 from the GTK port, which is the only remaining one using
libsoup2. This in turn allows to remove all the USE(SOUP2) guards, and
the SOUP_CHECK_VERSION(2, ...) ones. As side effects, the USE_SOUP2
CMake option is gone, and the definition of ENABLE_SERVER_PRECONNECT
is moved into PlatformEnableGlib.h as it is always enabled for both the
WPE and GTK ports.

Canonical link: https://commits.webkit.org/301131@main
